core/templates/dev/head/AppInit.ts

```diff
@@ -16,13 +16,35 @@
  * @fileoverview File for initializing the main oppia module.
  */
 
-import { NgModule } from '@angular/core';
+import 'core-js/es7/reflect';
+import 'zone.js';
+
+import { NgModule, StaticProvider, Component } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
-import { StaticProvider } from '@angular/core';
+import { downgradeInjectable, downgradeComponent } from
+  '@angular/upgrade/static';
+
+@Component({
+  selector: 'service-bootstrap',
+  template: ''
+})
+export class ServiceBootstrapComponent {}
+
+import { ExplorationFeaturesService } from
```

Comment:
This list is going to get big really quickly, shall we move these to a separate file?

Reply:
Actually, the list won't be that big once modularity is introduced. Then we can move these to their respective modules.

Reply:
Also, this needs to be declared in a module, and we have just one module.

Reply:
Just to check, when are we planning to introduce modularity? I just want to make sure we aren't going for a strategy of "do all these conversions first" (leading to a large, unmaintainable list) and only then "introduce modularity"...

Reply:
Oh no, I am currently working on modularity and then only will proceed with more services.

Reply:
okay, in that case, I'm fine with having this one service go in like this as long as modularity is before the other service migration work.

Reply:
Hmm ... I am looking at this again, and TBH it is _really weird_ to have ExplorationFeaturesService be in AppInit.ts. I think that would be a regression and I'm not keen to have that in develop.

So, what I would suggest is: now that we know how to migrate a service without breaking things (actually, just to double-check, you have manually tested this functionality on localhost, right?) --  I think that we shouldn't merge this as-is, but start working on using similar patterns as in this PR to convert services while preserving modularity. (It also means that this PR has served its purpose, assuming that @kevinlee12 has no remaining comments.)

Does that sound good?

Reply:
I am not sure what do you mean by 'regression' here, because this isn't breaking things.

Yes I agree withthis being weird looking in Appinit.ts.

By 'preserving modularity' do you mean that I introduce modularity first and then this PR is to be merged?

Reply:
Re "regression": it seems like a step backwards for the main file to import a random service file in the app. We should not be checking that in.

I'm happy to merge a PR that migrates services without importing things in places which they probably shouldn't be imported. If this involves introducing modularity first, then let's go ahead with that. But this PR should not be merged as-is, because the Appinit.ts import sets a pattern which we don't want others to follow.